integrations/helm/chartsync/chartsync.go

```diff
@@ -309,6 +315,72 @@ func (chs *ChartChangeSync) getCustomResources() ([]ifv1.FluxHelmRelease, error)
 	return fhrs, nil
 }
 
+func sortStrings(ss []string) []string {
+	ret := append([]string{}, ss...)
+	sort.Strings(ret)
+	return ret
+}
+
+type anySlice []*google_protobuf.Any
+
+func (a anySlice) Less(i, j int) bool {
+	return seq.Compare(a[i], a[j]) < 0
+}
+func (a anySlice) Len() int      { return len(a) }
+func (a anySlice) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+type templateSlice []*hapi_chart.Template
+
+func (t templateSlice) Less(i, j int) bool {
+	return seq.Compare(t[i], t[j]) < 0
+}
+func (t templateSlice) Len() int      { return len(t) }
+func (t templateSlice) Swap(i, j int) { t[i], t[j] = t[j], t[i] }
+
+type maintainerSlice []*hapi_chart.Maintainer
+
+func (m maintainerSlice) Less(i, j int) bool {
+	return seq.Compare(m[i], m[j]) < 0
+}
+func (m maintainerSlice) Len() int      { return len(m) }
+func (m maintainerSlice) Swap(i, j int) { m[i], m[j] = m[j], m[i] }
+
+type chartSlice []*hapi_chart.Chart
+
+func (a chartSlice) Less(i, j int) bool {
+	return seq.Compare(a[i], a[j]) < 0
+}
+func (a chartSlice) Len() int      { return len(a) }
+func (a chartSlice) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func sortChartFields(c *hapi_chart.Chart) *hapi_chart.Chart {
+	nc := hapi_chart.Chart{
+		Metadata:  &(*c.Metadata),
+		Templates: append([]*hapi_chart.Template{}, c.Templates...),
+		Files:     append([]*google_protobuf.Any{}, c.Files...),
+	}
+
+	if c.Values != nil {
+		nc.Values = &(*c.Values)
+	}
+
+	sort.Sort(anySlice(nc.Files))
+	sort.Sort(templateSlice(nc.Templates))
+
+	nc.Metadata.Sources = sortStrings(nc.Metadata.Sources)
```

Comment:
I also forgot about sort.Strings().  In this case though I probably still would've avoided it because I didn't want to mutate the given value.  Why be concerned about it you ask?  Habit I guess.   I agree in this case it doesn't likely matter, but I feel unless the cost is significant it's a good practice not to modify your inputs.

Reply:
That is a pretty sound reason. I looked through the docs for go-cmp and notice that many of the extension points (options) have the contract of "don't mutate the underlying value" -- seems reasonable to use the same contract here, even if not using those extension points.